github/client.go

```diff
@@ -970,6 +985,29 @@ func (client *Client) apiClient() *simpleClient {
 	return &simpleClient{
 		httpClient: httpClient,
 		rootUrl:    apiRoot,
+		extraHeader: func(r *http.Header) {
+			if headers, err := git.ConfigAll(fmt.Sprintf("http.%s://%s.extraheader", client.Host.Protocol, client.Host.Host)); err == nil {
```

Comment:
I'm wondering whether it would be better to design this feature so that `extraHeader` is configured from the "outside" (being passed the list of extra headers), rather than making git requests here. I want to avoid the HTTP client being unnecessarily coupled to git config.

Reply:
We have utility, to handel the process with KeyCloack as IDP.
This tool is writing the git/extraHeader for the git cli workflow.

https://github.com/git/git/blob/042ed3e048af08014487d19196984347e3be7d1c/Documentation/config/http.txt#L52

I could refactor this function, that it is call once and only the "r" manipulation is running